src/qtmaximise.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef QT_MAXIMISE_H_INCLUDED
+#define QT_MAXIMISE_H_INCLUDED
+
+#include <QAction>
+#include <QMainWindow>
+
+class QActionMaximise : public QAction
+{
+    Q_OBJECT
+
+    public:
+        QActionMaximise(QMainWindow* mainWindow = 0) : QAction(tr("Maximise"),mainWindow), m_mainWindow(mainWindow)
+        {
+            setShortcut(Qt::Key_F11);
+            connect(this, SIGNAL(triggered(bool)), this, SLOT(handleTriggered(bool)));
+        }
+
+    public slots:
+        void handleTriggered(bool checked)
+        {
+            const bool maximised = m_mainWindow->windowState()&Qt::WindowMaximized;
+            if (maximised)
+                m_mainWindow->showNormal();
+            else
+                m_mainWindow->showMaximized();
+        }
```

Comment:
Concerning QActionMaximise: Not sure it's worth quibbling over. In the past I've had various QAction collections for populating right-click context menus, rather than wiring those directly into QMainWindow by default, which can wind up being monolithic.  I would have liked QActionMaximise to be more context-sensitive - such as disabling itself when the window is already maximised via the window manager.  But I retreated on that when I ran into complications.  It would be easier to tackle that in isolation to the main window.

A broader issue this raises in my mind is possibly separating the core "model" and "rendering" from the broader GUI stuff.  Aside from other branches and forks the code seems due for a bit of a shuffle?